portfolio/src/main/webapp/comments.js

```diff
@@ -80,6 +93,19 @@ class Comments {
       throw Error(`${response.statusText}. Status: ${response.status}`);
     }
   }
+
+  /**
+   * Returns an error if the response status is not between 200 and 299, i.e not OK.
+   * @param {*} response - The HTTP response received from the servlet.
+   * @return {*} - If an error is not thrown, returns the response (not JSON).
+   */
+  checkPostError(response) {
+    if (response.ok) {
+      return;
+    } else {
+      throw Error(`${response.statusText}. Status: ${response.status}`);
+    }
```

Comment:
If you're not doing anything if the response is ok, there's no need to check for it. Simplify this block to 

```
if (!response.ok()) { ... }
```